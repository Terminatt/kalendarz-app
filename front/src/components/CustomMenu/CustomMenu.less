@import '~styles/variables.less';

.ant-menu.custom-menu {
  border: none;

  .ant-menu-submenu-title:focus {
    background-color: lighten(@secondary-bg, 20%);
  }
  
  .ant-menu-item.ant-menu-item-selected {
    background-color: lighten(@secondary-bg, 20%);
    color: @text-color;
  }

  .ant-menu-item:active, .ant-menu-submenu-title:active {
    background-color: @secondary-bg;
  }
}