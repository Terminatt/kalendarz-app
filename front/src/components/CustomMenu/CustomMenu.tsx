import { joinClassNames } from '@utils/general';
import { Menu, MenuProps } from 'antd';
import React from 'react';

import './CustomMenu.less';

const CustomMenu: React.FC<MenuProps> = (props) => {
    const { className, children, ...rest } = props;

    return <Menu {...rest} className={joinClassNames(['custom-menu', className])}>{children}</Menu>;
};

export default CustomMenu;
