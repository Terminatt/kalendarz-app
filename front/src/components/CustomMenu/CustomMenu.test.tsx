import React from 'react';
import renderer from 'react-test-renderer';
import CustomMenu from './CustomMenu';
import { Menu } from 'antd';

describe('Custom Menu Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomMenu>
                    <Menu.Item key="test">
                        Test
                    </Menu.Item>
                </CustomMenu>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});
