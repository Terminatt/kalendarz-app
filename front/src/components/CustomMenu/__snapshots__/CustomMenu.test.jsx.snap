// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Menu Component matches the snapshot 1`] = `
Array [
  <ul
    className="ant-menu ant-menu-root ant-menu-vertical ant-menu-light custom-menu"
    data-menu-list={true}
    onKeyDown={[Function]}
    role="menu"
    tabIndex={0}
  >
    <li
      className="ant-menu-item ant-menu-item-only-child"
      data-menu-id={null}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        Test
      </span>
    </li>
  </ul>,
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />,
]
`;
